### PR TITLE
ports the admin_delete proc

### PR DESCRIPTION
**Pull request by magatsuchi**
## About The Pull Request

replaces `atom_delete` with `admin_delete`, which allows turf deletions upto the base turf (which still can't be removed)

also removes a null-caused stack_trace with the `getShuttle` proc because that was getting annoying

## Why It's Good For The Game

qol + admins should be able to delete walls

## Changelog
:cl:
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
What does this actually do? I was looking in discord, but I see admins delete turfs all the time without issue.

**Comment by Lumipharon** [comment]
What does this actually do sorry? I was looking in discord, but I see admins delete turfs all the time. Or are they just doing a funni?

**Comment by magatsuchi** [comment]
> What does this actually do sorry? I was looking in discord, but I see admins delete turfs all the time. Or are they just doing a funni?

they're using buildmode, which is a viable option, but the admin delete procs should be capable of scraping away tiles as well. this doesn't give admins the ability to make space tiles: it just means they can scrape away them now, down to the base turf, which is not space in most instances.

File: code/modules/admin/view_variables/admin_delete.dm [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/client/proc/admin_delete(datum/delete)
```
Comment (TiviPlus) [inline code/modules/admin/view_variables/admin_delete.dm:1]: Missing autodoc

File: code/modules/admin/view_variables/admin_delete.dm [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	if(istype(atom_delete))
+		var/turf/turf_delete = get_turf(atom_delete)
+		if(turf_delete)
+			coords = "at [COORD(turf_delete)]"
+			jmp_coords = "at [ADMIN_COORDJMP(turf_delete)]"
+		else
+			jmp_coords = coords = "in nullspace"
+
+	if (tgui_alert(usr, "Are you sure you want to delete:\n[delete]\n[coords]?", "Confirmation", list("Yes", "No")) == "Yes")
+		log_admin("[key_name(usr)] deleted [delete] [coords]")
```
Comment (TiviPlus) [inline code/modules/admin/view_variables/admin_delete.dm:14]: use early returns

File: code/modules/admin/debug_verbs.dm [hunk 3]
```diff
@@ -678,3 +656,12 @@ GLOBAL_PROTECT(AdminProcCallSpamPrevention)
...
 
 	usr << browse(dellog.Join(), "window=dellog")
+
+/client/proc/cmd_admin_delete(atom/A as obj|mob|turf in world)
+	set category = "Debug"
+	set name = "Delete"
+
+	if(!check_rights(R_SPAWN|R_DEBUG))
+		return
+
```
Comment (TiviPlus) [inline code/modules/admin/debug_verbs.dm:666]:
```suggestion
	if(!check_rights(R_DEBUG))
		return

```
incorrect perms

File: code/modules/admin/holder.dm [hunk 4]
```diff
@@ -345,13 +345,13 @@ GLOBAL_PROTECT(admin_verbs_asay)
...
 	/datum/admins/proc/debug_mob_lists,
-	/datum/admins/proc/delete_atom,
 	/datum/admins/proc/SDQL2_query,
 	/datum/admins/proc/restart_controller,
 	/datum/admins/proc/check_contents,
 	/datum/admins/proc/reestablish_db_connection,
 	/datum/admins/proc/view_runtimes,
-	/client/proc/toggle_cdn
+	/client/proc/toggle_cdn,
+	/client/proc/cmd_admin_delete
```
Comment (TiviPlus) [inline code/modules/admin/holder.dm:354]:
```suggestion
	/client/proc/cmd_admin_delete,
```

File: code/modules/admin/view_variables/admin_delete.dm [hunk 5]
```diff
@@ -0,0 +1,24 @@
+/client/proc/admin_delete(datum/delete)
+	var/atom/atom_delete = delete
+	var/coords = ""
```
Comment (TiviPlus) [inline code/modules/admin/view_variables/admin_delete.dm:3]: Why is this completely missing any admin permissions/proccall/etc checks?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by magatsuchi** [comment]
fuck

**Pull request closed without merging**